docker/android/Dockerfile

```diff
@@ -59,26 +62,81 @@ RUN echo "Preparing Gradle..." && \
 ENV GRADLE_HOME=/opt/gradle-3.5
 ENV PATH=$PATH:$GRADLE_HOME/bin
 
-RUN echo "Preparing Android SDK..." && \
-	mkdir -p /opt/android-sdk-linux && \
-	cd /opt/android-sdk-linux && \
-	curl -L -O https://dl.google.com/android/repository/sdk-tools-linux-3859397.zip && \
-	unzip sdk-tools-linux-3859397.zip
+# V1 --------------------------------------------------
 
-ENV ANDROID_HOME=/opt/android-sdk-linux
+# ENV ANDROID_HOME=/opt/android-sdk
+# ENV PATH $PATH:$ANDROID_HOME/tools
+# ENV PATH $PATH:$ANDROID_HOME/platform-tools
 
-RUN yes | /opt/android-sdk-linux/tools/bin/sdkmanager --licenses
+# RUN echo "Preparing Android SDK..." && \
+# 	mkdir -p $ANDROID_HOME && \
+# 	cd $ANDROID_HOME && \
+# 	curl -L -O https://dl.google.com/android/repository/sdk-tools-linux-3859397.zip && \
+# 	unzip sdk-tools-linux-3859397.zip && \
+# 	rm -rf sdk-tools-linux-3859397.zip
 
-# RUN /opt/android-sdk-linux/tools/bin/sdkmanager --update --verbose
-RUN /opt/android-sdk-linux/tools/bin/sdkmanager --verbose "build-tools;24.0.3" "platforms;android-10"
+# RUN yes | $ANDROID_HOME/tools/bin/sdkmanager --licenses
+
+# # RUN $ANDROID_HOME/tools/bin/sdkmanager --update --verbose
+# RUN $ANDROID_HOME/tools/bin/sdkmanager --verbose	\
+# 	"platform-tools"								\
+# 	"emulator"										\
+# 	"build-tools;25.0.3"							\
+# 	"platforms;android-19"							\
+# 	"system-images;android-19;default;armeabi-v7a"	\
+# 	"system-images;android-19;default;x86"
+
+# ENV LD_LIBRARY_PATH=$ANDROID_HOME/emulator/lib64/qt/lib:$ANDROID_HOME/emulator/lib64/gles_mesa
+
+# RUN DEBIAN_FRONTEND=noninteractive apt-get -qq install -y libxdamage1 libxfixes3 qemu-kvm cpu-checker supervisor
+
+
+
+# V2 --------------------------------------------------
+
+# TODO: move to packages at top
+RUN echo "Preparing Debian packages..." && \
+	apt-get -qq update && \
+	DEBIAN_FRONTEND=noninteractive apt-get -qq install -y \
+	wget supervisor
+
+RUN wget -qO- http://dl.google.com/android/android-sdk_r23-linux.tgz | \
+    tar xvz -C /usr/local/ && \
+    mv /usr/local/android-sdk-linux /usr/local/android-sdk && \
+    chown -R root:root /usr/local/android-sdk/
+
+# Add android tools and platform tools to PATH
+ENV ANDROID_HOME /usr/local/android-sdk
+ENV PATH $PATH:$ANDROID_HOME/tools
+ENV PATH $PATH:$ANDROID_HOME/platform-tools
+
+RUN ( sleep 4 && while [ 1 ]; do sleep 1; echo y; done ) | android update sdk --no-ui --force -a --filter \
+    build-tools-24.0.3,tools,platform-tools,android-19,extra-android-m2repository,sys-img-x86-android-19,sys-img-armeabi-v7a-android-19 && \
+    echo "y" | android update adb
+
+# Create fake keymap file
+RUN mkdir /usr/local/android-sdk/tools/keymaps && \
+    touch /usr/local/android-sdk/tools/keymaps/en-us
+
+EXPOSE 22
```

Comment:
I got those parts from this [Dockerfile running the Android emulator](https://github.com/tracer0tong/android-emulator#how-to-connect-to-emulator).
The ports expose the ADB for interaction with the emulator from the outside, and the QEMU VNC port
(you can connect to the emulator using a VNC client and see the Android screen, touch inputs do not work though)